src/PointwiseFunctions/AnalyticSolutions/GrMhdSolutions/ConstantFlow.hpp

```diff
@@ -0,0 +1,182 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <array>
+#include <limits>
+#include <pup.h>
+
+#include "DataStructures/Tensor/TypeAliases.hpp"
+#include "Evolution/Systems/GrMhd/Tags.hpp"  // IWYU pragma: keep
+#include "Options/Options.hpp"
+#include "PointwiseFunctions/GeneralRelativity/GrTags.hpp"
+#include "Utilities/MakeArray.hpp"  // IWYU pragma: keep
+#include "Utilities/TMPL.hpp"
+#include "Utilities/TaggedTuple.hpp"
+
+namespace grmhd {
+namespace Solutions {
+
+/*!
+ * \brief Constant GrMhd flow in Cartesian coordinates
+ *
+ * An analytic solution to the 3-D GrMhd system. In Cartesian coordinates
+ * \f$(x, y, z)\f$, and using dimensionless units, the primitive quantities
+ * at a given time \f$t\f$ are then
+ *
+ * \f{align*}
+ * \rho(\vec{x},t) &= \rho, \\
+ * v_x(\vec{x},t) &= v_x,\\
+ * v_y(\vec{x},t) &= v_y,\\
+ * v_z(\vec{x},t) &= v_z,\\
+ * \epsilon(\vec{x}) &= \epsilon,\\
+ * \B_x(\vec{x},t) &= 0,\\
+ * \B_y(\vec{x},t) &= 0,\\
+ * \B_z(\vec{x},t) &= 0
+ * \f}
+ *
+ */
+class ConstantFlow {
+ public:
+  /// The rest mass density of the fluid.
+  struct RestMassDensity {
+    using type = double;
+    static constexpr OptionString help = {
+        "The rest mass density of the fluid."};
+    static type lower_bound() { return 0.0; }
+  };
+
+  /// The mean flow velocity.
+  struct MeanVelocity {
+    using type = std::array<double, 3>;
+    static constexpr OptionString help = {"The mean flow velocity."};
+  };
+
+  /// The specific internal energy of the fluid.
+  struct SpecificInternalEnergy {
+    using type = double;
+    static constexpr OptionString help = {
+        "The specific internal energy of the fluid."};
+  };
+
+  /// The lapse.
```

Comment:
Perhaps it should be documented that these 3+1 variables are not a solution to the Einstein equation using this flow as a source. These are constant numbers that make no impact on the right-hand side of the GRMHD equations, but we set them equal to something different from zero just for testing purposes.